x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -196,7 +222,16 @@ export function getIndexPatternDatasource(chrome: Chrome, toastNotifications: To
     },
 
     toExpression(state: IndexPatternPrivateState) {
-      return `${JSON.stringify(state.columns)}`;
+      if (state.columnOrder.length === 0) {
+        return '';
+      }
+
+      const fieldNames = state.columnOrder.map(col => state.columns[col].sourceField);
+      const expression = `esdocs index="${state.currentIndexPatternId}" fields="${fieldNames.join(
```

Comment:
I'm wondering if we need to escape the field names here. Is there a possibility that field names have weird characters, such as `"` which will cause the expression string to be invalid?

Reply:
Characters should already have been escaped during JSON deserialization, so I don't see the possible errors here.